src/integration/java/io/pinecone/integration/dataplane/UpdateAndQueryTest.java

```diff
@@ -0,0 +1,438 @@
+        List<String> upsertIds = Arrays.asList("v1", "v2", "v3");
+        String namespace = RandomStringBuilder.build("ns", 8);
+        blockingStub.upsert(buildRequiredUpsertRequest(upsertIds, namespace));
+        Thread.sleep(3500);
+
+        // Verify the upserted vector count with fetch
+        FetchRequest fetchRequest = FetchRequest.newBuilder().addAllIds(upsertIds).setNamespace(namespace).build();
+        FetchResponse fetchResponse = blockingStub.fetch(fetchRequest);
+        assertEquals(fetchResponse.getVectorsCount(), upsertIds.size());
+        for (String key : upsertIds) {
+            assert (fetchResponse.containsVectors(key));
+        }
+
+        // Update required fields only
+        UpdateRequest updateRequest = UpdateRequest.newBuilder()
+                .setId(upsertIds.get(0))
+                .setNamespace(namespace)
+                .addValues(100F)
+                .build();
+
+        // Should fail since only 1 value is added for the vector of dimension 3
+        try {
+            blockingStub.update(updateRequest);
+        } catch (StatusRuntimeException statusRuntimeException) {
+            assert (statusRuntimeException.getTrailers().toString().contains("grpc-status=3"));
+            assert (statusRuntimeException.getTrailers().toString().contains("Vector dimension 1 does not match the dimension of the index 3"));
+        }
+    }
+
+    @Test
+    public void UpsertRequiredParamsVectorAndQueryByIncorrectVectorDimensionSync() throws InterruptedException {
+        // Upsert vectors with required parameters
+        List<String> upsertIds = Arrays.asList("v1", "v2", "v3");
+        String namespace = RandomStringBuilder.build("ns", 8);
+        try {
+            blockingStub.upsert(buildRequiredUpsertRequest(upsertIds, namespace));
+            Thread.sleep(3500);
+
+            QueryRequest queryRequest = QueryRequest.newBuilder()
+                    .setNamespace(namespace)
+                    .setTopK(5)
+                    .setIncludeValues(true)
+                    .setIncludeMetadata(true)
+                    .addVector(100F)
+                    .build();
+
+            blockingStub.query(queryRequest);
+        } catch (StatusRuntimeException statusRuntimeException) {
+            assert (statusRuntimeException.getTrailers().toString().contains("grpc-status=3"));
+            assert (statusRuntimeException.getTrailers().toString().contains("grpc-message=Query vector dimension 1 does not match the dimension of the index 3"));
+        }
+    }
+
+    @Test
+    public void UpdateRequiredParamsFetchAndQueryFuture() throws InterruptedException, ExecutionException {
+        // Upsert vectors with required parameters
+        List<String> upsertIds = Arrays.asList("v1", "v2", "v3");
+        String namespace = RandomStringBuilder.build("ns", 8);
+        futureStub.upsert(buildRequiredUpsertRequest(upsertIds, namespace));
+        Thread.sleep(3500);
```

Comment:
Nitpick, but it might be nice to set the `3500` to a `final` constant that we use across the file so we can adjust things more easily if needed.